receiver/prometheusexecreceiver/subprocessmanager/manager.go

```diff
@@ -80,23 +64,40 @@ func (proc *Process) Run(logger *zap.Logger) (time.Duration, error) {
 	go proc.pipeSubprocessOutput(bufio.NewReader(stderrReader), logger, false)
 
 	// Start and stop timer (elapsed) right before and after executing the command
+	done := make(chan error, 1)
 	start := time.Now()
+
 	errProcess := childProcess.Start()
 	if errProcess != nil {
 		return 0, fmt.Errorf("process could not start: %w", errProcess)
 	}
 
-	errProcess = childProcess.Wait()
-	elapsed := time.Since(start)
-	if errProcess != nil {
-		return elapsed, fmt.Errorf("process error: %w", errProcess)
-	}
+	go func() {
+		done <- childProcess.Wait()
+	}()
+
+	// Handle normal process exiting or parent logic triggering a shutdown
+	select {
+	case errProcess := <-done:
+		elapsed := time.Since(start)
 
-	return elapsed, nil
+		// The subprocess will most likely exit due to an error (which is logged), but in any case we'll continue as usual with the health/delay computation
+		if errProcess != nil {
+			return elapsed, fmt.Errorf("process error: %w", errProcess)
+		}
+		return elapsed, nil
+
+	case <-ctx.Done():
+		err := childProcess.Process.Kill()
```

Comment:
You should probably only revert to this on non-Linux OSs, as on Linux `Pdeathsig` will take care of this more gracefully?

Reply:
`Pdeathsig` will take care of it better, but what if `Shutdown` is not user-induced, and instead Collector-triggered for something else, for example a config change!? Then I need to explicitly kill the process, no?

Reply:
Yes good point

Reply:
Is there any point doing `Pdeathsig` then? That seems redundant now so I would remove it.

Also, this is probably not critical for now, but it would be better to attempt a SIGTERM (not possible on Windows) before falling back to just killing the process